src/YieldSpaceAMO.sol

```diff
@@ -0,0 +1,465 @@
+    ) Owned(_ownerAddress) {
+        FRAX = IFrax(0x853d955aCEf822Db058eb8505911ED77F175b99e);
+        amoMinter = IFraxAMOMinter(_amoMinterAddress);
+        timelockAddress = amoMinter.timelock_address();
+
+        ladle = ILadle(_yieldLadle);
+        cauldron = ICauldron(ladle.cauldron());
+        fraxJoin = _yieldFraxJoin;
+
+        currentAMOmintedFRAX = 0;
+        currentAMOmintedFyFRAX = 0;
+    }
+
+    /* ============== MODIFIERS ============== */
+    modifier onlyByOwnGov() {
+        require(
+            msg.sender == timelockAddress || msg.sender == owner,
+            "Not owner or timelock"
+        );
+        _;
+    }
+
+    modifier onlyByMinter() {
+        require(msg.sender == address(amoMinter), "Not minter");
+        _;
+    }
+
+    /* ================ VIEWS ================ */
+    // /// @notice returns current rate on Frax debt
+    // function getRate() public view returns (uint256) { //TODO Name better & figure out functionality
+    //     return (circulatingAMOMintedFyFrax() - currentRaisedFrax()) / (currentRaisedFrax() * /*timeremaining*/; //TODO pos/neg
+    // }
+
+    function showAllocations(
+        bytes6 seriesId
+    ) public view returns (uint256[6] memory return_arr) {
+        Series storage _series = series[seriesId];
+        require(_series.vaultId != bytes12(0), "Series not found");
+
+        uint256 fraxInContract = FRAX.balanceOf(address(this));
+        uint256 fraxAsCollateral = cauldron.balances(_series.vaultId).ink;
+        uint256 fraxInLP = (FRAX.balanceOf(address(_series.pool)) *
+            _series.pool.balanceOf(address(this))) / _series.pool.totalSupply();
+        uint256 fyFraxInContract = _series.fyToken.balanceOf(address(this));
+        uint256 fyFraxInLP = (_series.fyToken.balanceOf(address(_series.pool)) *
+            _series.pool.balanceOf(address(this))) / _series.pool.totalSupply();
+        uint256 LPOwned = _series.pool.balanceOf(address(this));
+        return [
+            fraxInContract, // [0] Unallocated Frax
+            fraxAsCollateral, // [1] Frax being used as collateral to borrow fyFrax
+            fraxInLP, // [2] The Frax our LP tokens can lay claim to
+            fyFraxInContract, // [3] fyFrax sitting in AMO, should be 0
+            fyFraxInLP, // [4] fyFrax our LP can claim
+            LPOwned // [5] number of LP tokens
+        ];
+    }
+
+    /// @notice Return the Frax value of a fyFrax amount, considering a debt repayment if possible.
+    function fraxValue(
+        bytes6 seriesId,
+        uint256 fyFraxAmount
+    ) public view returns (uint256 fraxAmount) {
+        Series storage _series = series[seriesId];
+        uint256 debt = cauldron.balances(series[seriesId].vaultId).art;
+        if (debt > fyFraxAmount) {
+            fraxAmount = fyFraxAmount;
+        } else {
+            fraxAmount =
+                debt +
+                _series.pool.sellFYTokenPreview((fyFraxAmount - debt).u128());
```

Comment:
what happens if sellFYTokenPreview() is provided a value greater than it can trade?

Reply:
This would revert. I added a TODO, and will write about the conditions in which that would happen.